devito/function.py

```diff
@@ -812,14 +840,14 @@ def __indices_setup__(cls, **kwargs):
     def __shape_setup__(cls, **kwargs):
         grid = kwargs.get('grid')
         dimensions = kwargs.get('dimensions')
-        shape = kwargs.get('shape')
+        shape = kwargs.get('shape', kwargs.get('shape_global'))
```

Comment:
that look odd to me a `get` inside a `get`

Reply:
nesting ``kwargs.get`` essentially means ``A or B or C or ... or None``. It's fine